elastic/types.go

```diff
@@ -0,0 +1,323 @@
+package elastic
+
+// Negate wraps a term in a boolean "not" condition
+func Negate(t Term) Term {
+	b := Term{
+		"bool": Term{
+			"not": t,
+		},
+	}
+	return b
+}
+
+// Nest wraps a term in a "nested" statement
+func Nest(t Term) Term {
+	n := Term{
+		"nested": t,
+	}
+	return n
+}
+
+// Query wraps a term in a "query" statement
+func Query(t Term) Term {
+	q := Term{
+		"query": t,
+	}
+	return q
+}
+
+// Filter wraps a term in a "filter" statement
+func Filter(t Term) Term {
+	f := Term{
+		"filter": t,
+	}
+	return f
+}
+
+// Term is an alias for map[string]interface{}
+type Term map[string]interface{}
```

Comment:
In general, it reads a bit easier to have types defined first in the file (below any constants and variables) since most functions will depend on those types.